atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1462,14 +1463,25 @@ private void deleteOrphanedSentinelsAsync(TableReference table, Set<Cell> actual
                         .map(_ignore -> Value.INVALID_VALUE_TIMESTAMP)
                         .collectToSetMultimap();
                 try {
-                    deleteExecutor.execute(() -> keyValueService.delete(table, sentinels));
+                    runTaskOnDeleteExecutor(kvs -> kvs.delete(table, sentinels));
                 } catch (Throwable th) {
                     // best effort
                 }
             }
         });
     }
 
+    /**
+     * This method exists to ensure that tasks run on the delete executor do not have references to the
+     * SnapshotTransaction that spawned the task (so that it can be garbage collected, which may be significant as
+     * large write or serializable read transactions may have allocated a lot of memory).
+     */
+    private void runTaskOnDeleteExecutor(Consumer<KeyValueService> task) {
```

Comment:
Pretty smart :)
Very nice find, this must have been a nightmare to figure out 🗡️